packages/devtools_app/test/test_infra/scenes/standalone_ui/utils.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:stream_channel/stream_channel.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+/// Connects to the websocket at [wsUri] and returns a channel along with
+/// a log stream that includes all protocol traffic.
+Future<({StreamChannel<String> channel, Stream<String> log})>
+    createLoggedWebSocketChannel(Uri wsUri) async {
+  final logController = StreamController<String>();
+
+  final rawChannel = WebSocketChannel.connect(wsUri);
```

Comment:
does this channel need to be closed at some point by the caller of createLoggedWebSocketChannel? Otherwise we might be leaking resources

Reply:
I believe this is the same as above. The stream gets passed to the `DartToolingDaemon` class and it will handle closing them (which is handled by the fake editor if you click the disconnect button, but in this test app we don't really have another way to close it - eg. you can't navigate away from the mock editor scene when you run this app).